src/core/algorithm/timelinePagerank.js

```diff
@@ -166,12 +163,12 @@ export function* _timelineNodeWeights(
   }
 }
 
-export function* _timelineMarkovChain(
+export function* _timelineNodeToConnections(
   graph: Graph,
   edgeCreationHistory: $ReadOnlyArray<$ReadOnlyArray<Edge>>,
   edgeEvaluator: EdgeWeightEvaluator,
   intervalDecay: number
-): Iterator<SparseMarkovChain> {
+): Iterator<NodeToConnections> {
```

Comment:
Nit: Suggest using `Generator<NodeToConnections, void, void>` as a return type.
As (Async)Generators implement both `Iterator` and `Iterable`, it allows for more flexible usage.

Function arguments should remain specific though. For example when using the argument in a for..of loop, it should be an `Iterable<T>`, even when passing in a Generator. This makes it easier to create a source of data. For example in a unit test, array literals like `[1, 2, 3]` can be used as they implement `Iterable<Number>` but aren't Generators.

Reply:
I don't see any practical benefit to doing this--can you give an example of a plausible situation where we'd be glad for this change? In general, I think adding complexity for purely hypothetical benefits is an antipattern.